shared/reports/totals.py

```diff
@@ -0,0 +1,59 @@
+from typing import Iterator
+
+from shared.helpers.numeric import ratio
+from shared.reports.types import ReportLine, ReportTotals
+from shared.utils.merge import LineType, line_type
+
+
+def get_line_totals(lines: Iterator[ReportLine]) -> ReportTotals:
+    """
+    Calculates the totals (`ReportTotals`) across all the given `lines` (`ReportLine`s).
+    """
+    hits = 0
+    misses = 0
+    partials = 0
+    branches = 0
+    methods = 0
+    messages = 0
+    complexity = 0
+    complexity_total = 0
+
+    for line in lines:
+        match line_type(line.coverage):
+            case LineType.hit:
+                hits += 1
+            case LineType.miss:
+                misses += 1
+            case LineType.partial:
+                partials += 1
+
+        if line.type == "b":
+            branches += 1
+        elif line.type == "m":
+            methods += 1
+
+        if line.messages:
```

Comment:
what is this?

Reply:
A completely unused thing :-D Otherwise I don’t think anyone really knows.